hoomd/CellListGPU.cc

```diff
@@ -24,22 +24,15 @@ CellListGPU::CellListGPU(std::shared_ptr<SystemDefinition> sysdef)
         throw std::runtime_error("Error initializing CellListGPU");
         }
 
-    unsigned int max_threads = m_exec_conf->dev_prop.maxThreadsPerBlock;
-    unsigned int warp_size = m_exec_conf->dev_prop.warpSize;
-    m_tuner.reset(new Autotuner(warp_size,
-                                max_threads,
-                                warp_size,
-                                5,
-                                100000,
-                                "cell_list",
-                                this->m_exec_conf));
-    m_tuner_combine.reset(new Autotuner(warp_size,
-                                        max_threads,
-                                        warp_size,
-                                        5,
-                                        100000,
-                                        "cell_list_combine",
-                                        this->m_exec_conf));
+    m_tuner.reset(new Autotuner<1>({AutotunerInterface::makeBlockSizeRange(m_exec_conf)},
+                                   this->m_exec_conf,
+                                   "cell_list"));
+    m_autotuners.push_back(m_tuner);
+
+    m_tuner_combine.reset(new Autotuner<1>({AutotunerInterface::makeBlockSizeRange(m_exec_conf)},
+                                           this->m_exec_conf,
+                                           "cell_list_combine"));
+    m_autotuners.push_back(m_tuner_combine);
```

Comment:
We could, yes. Would that set their string name as well? It would be the most logical connection between the Python dict of tuples interface. However if that were the case, how would they know their name to include in the notice messages?

Reply:
Hmm, that is a good point. We could provide an interface for adding or removing tuners from the base class that set's it both places.

Reply:
Yeah, if the notice messages are necessary then the name has to live in two places. I was trying to actually reduce the code complexity (from a human perspective not performance). Having to store every auto tuner in two places seems a bad pattern. We can't just use a `std::vector` though because that would obfuscate the code which is why I suggested an `unordered_map`. The slow down would only be while tuning correct? If so that shouldn't effect performance as a whole by much.

Reply:
> The slow down would only be while tuning correct? If so that shouldn't effect performance as a whole by much.
No, the tuner's `begin` and `end` methods are called for every kernel launch. The caller doesn't know whether the tuner is active or not.